scripts/ci/run-test-msbuild.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -u
+set -e
+# set -x
+
+OLD_CWD=`pwd`
+TESTCMD=`realpath ${TESTCMD}`
+
+${TESTCMD} --label=remove-mono-temp-dir --timeout=1m rm -rf /tmp/xplat-master
+${TESTCMD} --label=autogen --timeout=15m ./autogen.sh --prefix=/tmp/mono-from-source
+${TESTCMD} --label=make --timeout=30m make PROFILE=net_4_x -j 2
+${TESTCMD} --label=make-install --timeout=15m make install
+
+OLD_PATH=${PATH}
+export PATH=/tmp/mono-from-source/bin/:${OLD_PATH}
+${TESTCMD} --label=cert-sync --timeout=15m cert-sync /etc/ssl/certs/ca-certificates.crt
+
+rm -rf /tmp/xplat-master || true
+
+MSBUILD_REVISION=`grep -Po "(?<=revision \= \')[0-9a-f]+" packaging/MacSDK/msbuild.py`
+echo Detected msbuild hash ${MSBUILD_REVISION} in MacSDK/msbuild.py
+
+${TESTCMD} --label=clone-msbuild --timeout=5m git clone https://github.com/mono/msbuild.git /tmp/xplat-master
+cd /tmp/xplat-master
+${TESTCMD} --label=checkout-xplat-master --timeout=5m git checkout ${MSBUILD_REVISION}
+
+${TESTCMD} --label=compile-msbuild --timeout=15m ./eng/cibuild_bootstrapped_msbuild.sh --host_type mono --configuration Release /p:DisableNerdbankVersioning=true /p:CreateBootstrap=true "/p:Projects=/tmp/xplat-master/src/MSBuild.Bootstrap/MSBuild.Bootstrap.csproj" /p:AssemblyVersion=15.1.0.0
+
+MONO=/tmp/mono-from-source/bin/mono
+DLL_PATH=`realpath /tmp/xplat-master/artifacts/2/bin/MSBuild.Bootstrap/*-MONO/net472/MSBuild.dll`
```

Comment:
It would be much better to install msbuild to `/tmp/mono-from-source` and then use `DLL_PATH=/tmp/mono-from-source/lib/mono/msbuild/Current/bin/MSBuild.dll`.
msbuild is not just the stuff in that directory.

Reply:
OK, I'll work on trying to do that instead

Reply:
The test script does a proper install to the prefix and tests that, so I think that's enough. The old-style test is still in here but that's because it seems like some customers do it. If it breaks here that's a good reason to remove it from the test and update the documentation to clearly tell them not to do it. :) Maybe we should update it anyway?